codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -282,6 +284,16 @@ public void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
                 return;
             }
 
+            boolean isInformational = false;
+            if ((!connection.isServer() &&
+                    (isInformational = HttpStatusClass.valueOf(headers.status()) == INFORMATIONAL)
+                    && stream.isHeadersReceived()) ||
+                stream.isTrailersReceived() || stream.isHeadersReceived() && !endOfStream) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'stream' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2259)